### ISSUE TITLE
Provide header information on /containers/{id}/archive

### ISSUE THREAD

**Issue by sulksamino** [fussybeaver/bollard#487]
TLDR:  Can we get the header information on responses to `/containers/{id}/archive`?

We successfully use `bollard::container::Docker::download_from_container` to download files or directories from containers. In some cases we need to know if the path in the container points to a file or directory. Additionally it would be nice to know the size of the downloaded archive beforehand.

The [Docker Engine API](https://docs.docker.com/reference/api/engine/version/v1.46/#tag/Container/operation/ContainerArchiveInfo) mentions the response header X-Docker-Container-Path-Stat with a base64 encoded json object containing file system information being returned on a HEAD request on `/containers/{id}/archive` (a GET request seems to return the same header). This seems to correspond to the [following struct](https://github.com/moby/moby/blob/9552706f457625cc5b4a85c2cc89d67feafbc763/api/types/container/container.go#L23):
```go
type PathStat struct {
	Name       string      `json:"name"`
	Size       int64       `json:"size"`
	Mode       os.FileMode `json:"mode"`
	Mtime      time.Time   `json:"mtime"`
	LinkTarget string      `json:"linkTarget"`
}
```

Is there currently any way to get this information via bollard? If not would it be possible to implement?

Thank you for creating bollard

**Comment by fussybeaver** [comment]
It sounds like you may need to implement that as a new method, I don't see that struct in the generated openapi models either. Happy to have that contributed!
